frontend/src/app/commonComponents/MultiSelect/MultiSelect.tsx

```diff
@@ -0,0 +1,182 @@
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import _ from "lodash";
+
+import Required from "../Required";
+import Optional from "../Optional";
+
+import MultiSelectDropdown, {
+  MultiSelectDropdownOption,
+} from "./MultiSelectDropdown/MultiSelectDropdown";
+
+import "./MultiSelect.scss";
+
+export type MultiSelectProps = {
+  name: string;
+  label: React.ReactNode;
+  labelSrOnly?: boolean;
+  required?: boolean;
+  errorMessage?: React.ReactNode;
+  validationStatus?: "error" | "success";
+  ariaDescribedBy?: string;
+  hintText?: string | React.ReactNode;
+  onChange: (selectedItems: string[]) => void;
+  labelClassName?: string;
+  options: MultiSelectDropdownOption[];
+  className?: string;
+  initalSelectedOptions?: Array<string>;
+  disabled?: boolean;
+  inputProps?: JSX.IntrinsicElements["input"];
+};
+
+type PillProps = {
+  option: MultiSelectDropdownOption;
+  onDelete: (option: MultiSelectDropdownOption) => void;
+};
+
+const Pill = (props: PillProps) => (
+  <div className="pill">
+    {props.option.label}
+    <div
+      className="close-button"
+      data-testid={`${props.option.label}-pill-delete`}
+      onClick={() => props.onDelete(props.option)}
+    >
+      <FontAwesomeIcon fontSize={24} icon={"times"} />
+    </div>
+  </div>
+);
+
+const getSortedOptions = (options: MultiSelectDropdownOption[]) =>
+  _.orderBy(options, ["label"], ["asc"]);
+
+export const MultiSelect = ({
+  name,
+  label,
+  labelSrOnly,
+  errorMessage,
+  className,
+  required,
+  validationStatus,
+  ariaDescribedBy,
+  hintText,
+  onChange,
+  labelClassName,
+  options,
+  disabled,
+}: MultiSelectProps): React.ReactElement => {
+  const isDisabled = !!disabled;
+
+  const [availableOptions, setAvailableOptions] = useState<
+    MultiSelectDropdownOption[]
+  >(getSortedOptions(options));
+  const [selectedItems, setSelectedItems] = useState<string[] | undefined>(
+    undefined
+  );
+
+  const onItemSelected = (option: MultiSelectDropdownOption) => {
+    const newSelectedItems = selectedItems
+      ? [...selectedItems, option.value]
+      : [option.value];
+    setSelectedItems(Array.from(new Set(newSelectedItems)));
+    setAvailableOptions(
+      availableOptions.filter((_option) => _option.value !== option.value)
+    );
+  };
+
+  const onItemUnSelected = (option: MultiSelectDropdownOption) => {
+    const selectedItemsSet = new Set(selectedItems);
+    selectedItemsSet.delete(option.value);
+    setSelectedItems(Array.from(selectedItemsSet));
+    const sortedOptions = getSortedOptions([...availableOptions, option]);
+    setAvailableOptions(sortedOptions);
+  };
+
+  useEffect(() => {
+    const sortedOptions = getSortedOptions([...options]);
+    setAvailableOptions(sortedOptions);
+  }, [options, setAvailableOptions]);
+
+  useEffect(() => {
+    if (selectedItems) {
+      onChange(selectedItems);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
**suggestion:** Please add a comment explaining why `onChange` can't be in the dep list